nautobot/dcim/factory.py

```diff
@@ -0,0 +1,137 @@
+
+
+class PlatformFactory(OrganizationalModelFactory):
+    class Meta:
+        model = Platform
+        exclude = ("has_manufacturer", "manufacturer_slug", "has_description", "has_napalm_args")
+
+    # This dictates `name` and `napalm_driver`.
+    has_manufacturer = factory.Faker("pybool")
+
+    # Slug isn't defined here since it will always inherit from name.
+    name = factory.Maybe(
+        "has_manufacturer",
+        factory.LazyAttributeSequence(lambda o, n: f"{o.manufacturer.name} Platform {n + 1}"),
+        factory.Sequence(lambda n: f"Platform {n}"),
+    )
+
+    manufacturer = factory.Maybe("has_manufacturer", random_instance(Manufacturer), None)
+
+    # If it has a manufacturer, it *might* have a naplam_driver.
+    napalm_driver = factory.Maybe(
+        "has_manufacturer",
+        factory.LazyAttribute(lambda o: random.choice(NAPALM_DRIVERS.get(o.manufacturer.slug, [""]))),
+        "",
+    )
+
+    has_napalm_args = factory.Faker("pybool")
+    napalm_args = factory.Maybe(
+        "has_napalm_args", factory.Faker("pydict", nb_elements=2, value_types=[str, bool, int]), None
+    )
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
+
+
+class DeviceTypeFactory(PrimaryModelFactory):
+    class Meta:
+        model = DeviceType
+        exclude = (
+            "has_part_number",
+            "is_subdevice_child",
+            "has_comments",
+        )
+
+    manufacturer = random_instance(Manufacturer)
+
+    # Slug isn't defined here since it will always inherit from model.
+    model = factory.LazyAttributeSequence(lambda o, n: f"{o.manufacturer.name} DeviceType {n + 1}")
+
+    has_part_number = factory.Faker("pybool")
+    part_number = factory.Maybe("has_part_number", factory.Faker("ean", length=8), "")
+
+    # If randomly a subdevice, set u_height to 0.
+    is_subdevice_child = factory.Faker("boolean", chance_of_getting_true=33)
+    u_height = factory.Maybe("is_subdevice_child", 0, factory.fuzzy.FuzzyInteger(1, 2))
+
+    is_full_depth = factory.Faker("pybool")
+
+    # If randomly a subdevice, also set subdevice_role to "child". We might want to reconsider this.
+    subdevice_role = factory.Maybe(
+        "is_subdevice_child",
+        SubdeviceRoleChoices.ROLE_CHILD,
+        factory.Faker("random_element", elements=["", SubdeviceRoleChoices.ROLE_PARENT]),
+    )
+
+    has_comments = factory.Faker("pybool")
+    comments = factory.Maybe("has_comments", factory.Faker("paragraph"), "")
+
+
+class DeviceRoleFactory(OrganizationalModelFactory):
+    class Meta:
+        model = DeviceRole
+        exclude = ("has_description",)
+
+    # Slug isn't defined here since it will always inherit from name.
+    name = factory.LazyFunction(
+        lambda: "".join(word.title() for word in faker.Faker().words(nb=2, part_of_speech="adjective", unique=True))
+    )
```

Comment:
Do we want to follow the above pattern of 
```py
factory.Sequence(lambda n: f"Device Role {n}")
```

instead of random words? Perhaps `{unique adjective} Device Role`?

Reply:
Yeah I'll do that.